chain/tests/e2e/plasticcredit/project_tx_test.go

```diff
@@ -0,0 +1,364 @@
+				var resp plasticcredit.QueryProjectResponse
+				s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &resp))
+				s.Require().Equal(tc.expectedState, &resp.Project)
+			}
+		})
+	}
+}
+
+func (s *E2ETestSuite) TestCmdUpdateProject() {
+	val := s.network.Validators[0]
+
+	applicantKey, err := val.ClientCtx.Keyring.Key(applicantKey)
+	s.Require().NoError(err)
+	notApplicantKey, _ := val.ClientCtx.Keyring.Key(issuerKey)
+	s.Require().NoError(err)
+
+	testCases := map[string]struct {
+		args              []string
+		expectedErrOnSend bool
+		expectedErrOnExec bool
+		expectedErrMsg    string
+		expectedState     *plasticcredit.Project
+	}{
+
+		"update project": {
+			[]string{"3", "My Updated Project", fmt.Sprintf("--%s=%s", flags.FlagFrom, applicantKey.Name)},
+			false,
+			false,
+			"",
+			&plasticcredit.Project{
+				Id:                      3,
+				ApplicantId:             1,
+				CreditClassAbbreviation: "EMP",
+				Name:                    "My Updated Project",
+				Status:                  plasticcredit.ProjectStatus_APPROVED,
+			},
+		},
+		"update non-existing project": {
+			[]string{"65", "My Updated Project", fmt.Sprintf("--%s=%s", flags.FlagFrom, applicantKey.Name)},
+			false,
+			true,
+			"project not found",
+			nil,
+		},
+		"invalid applicant": {
+			[]string{"1", "My Updated Project", fmt.Sprintf("--%s=%s", flags.FlagFrom, notApplicantKey.Name)},
+			false,
+			true,
+			"unauthorized",
+			nil,
+		},
+		"empty name": {
+			[]string{"1", "", fmt.Sprintf("--%s=%s", flags.FlagFrom, applicantKey.Name)},
+			true,
+			false,
+			"project name cannot be empty",
+			nil,
+		},
+	}
+	for name, tc := range testCases {
+		s.Run(name, func() {
+			cmd := cli.MsgUpdateProjectCmd()
+			out, _ := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, append(tc.args, s.commonFlags...))
+			if tc.expectedErrOnSend {
+				s.Require().Contains(out.String(), tc.expectedErrMsg)
+			} else if tc.expectedErrOnExec {
+				var txResponse sdk.TxResponse
+				s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txResponse))
+				s.Require().Contains(txResponse.RawLog, tc.expectedErrMsg)
+			} else {
+				cmd = cli.CmdQueryProject()
+				out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, []string{tc.args[0]})
+				s.Require().NoError(err)
+				var resp plasticcredit.QueryProjectResponse
+				s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &resp))
+				s.Require().Equal(tc.expectedState, &resp.Project)
+			}
+		})
+	}
+}
+
+func (s *E2ETestSuite) TestCmdApproveProject() {
+	val := s.network.Validators[0]
+	issuerKey, err := val.ClientCtx.Keyring.Key(issuerKey)
+	s.Require().NoError(err)
+	notAdminKey, err := val.ClientCtx.Keyring.Key(applicantKey)
+	s.Require().NoError(err)
+	testCases := map[string]struct {
+		args              []string
+		expectedErrOnSend bool
+		expectedErrOnExec bool
+		expectedErrMsg    string
+		expectedState     *plasticcredit.Project
+	}{
+
+		"approve project": {
```

Comment:
```suggestion
		"approve new project": {
```
Just for clarification